pinot-core/src/main/java/com/linkedin/pinot/core/io/readerwriter/impl/FixedByteSingleColumnSingleValueReaderWriter.java

```diff
@@ -28,23 +29,101 @@
 // TODO Check if MMAP allows us to pack more partitions on a single machine.
 public class FixedByteSingleColumnSingleValueReaderWriter extends BaseSingleColumnSingleValueReaderWriter {
 
-  FixedByteSingleValueMultiColWriter currentWriter;
-  private List<FixedByteSingleValueMultiColWriter> writers = new ArrayList<>();
-  // For readers it is useful to keep a "chunked-reader" class that is aware of the start and end offset of that buffer.
-  // If we do that then we avoid the % operation on the record on every read.
-  // For the bulk read operation, we can do a loop like this one:
-  //   foreach(row in rows[]) {
-  //     bufferId = row / chunkSize;
-  //     chunked-readers.get(bufferId).getInt(row);
-  //   }
-  // TODO the logic above
-  private List<FixedByteSingleValueMultiColReader> readers = new ArrayList<>();
+  WriterWithOffset currentWriter;
+  private List<WriterWithOffset> writers = new ArrayList<>();
+  private List<ReaderWithOffset> readers = new ArrayList<>();
   private List<PinotDataBuffer> buffers = new ArrayList<>();
   private final long chunkSizeInBytes;
   private final int numRowsPerChunk;
   private final int columnSizesInBytes;
   private long numBytesInCurrentWriter = 0;
 
+  private static class WriterWithOffset implements Closeable {
```

Comment:
Sub class of what?

For now, I would like to leave this as inner class because it is not used anywhere else. As the implementation proceeds, we can evaluate how it turns out and pull it out if necessary. Agreed?